src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -140,6 +142,11 @@
      */
     private static final int GUARD_REGEN_INTERVAL = 40;
 
+    /**
+     * Interval between saturation losses during rallying. 1 saturation loss per interval.
+     */
+    private static final int RALLY_SATURATION_LOSS_INTERVAL = TICKS_SECOND * 10;
```

Comment:
Did you consider that they use up food too when fighting?

Reply:
I did. That is why the saturation reduction from this mechanism is always a fraction of your current saturation, ie. when you have low saturation, it won't deduct a lot, so it's not that punishing. Actually, this mechanism alone will only affect your saturation very little when you are below 3 points or so. The point of this code is mainly to force the saturation down to <2 after ~5 minutes of idling/walking, at which point unrallying will flatly set the saturation to 0, thereby forcing them to eat.

Reply:
kk nice :D was just making sure you considered it

Reply:
I changed the penalty a little bit in e54bba8. It will now reduce the saturation even slower (after about 5 minutes you reach LOW_SATURATION (which is 3 currently), but the flat reduction on unrallying is also increased to LOW_SATURATION. So, they fight longer, but will more likely have to eat after rallying.